### ISSUE TITLE
Bug in `obspy.taup.tau_model.TauModel.load_from_depth_cache` ?

### ISSUE THREAD

**Issue by rizac** [obspy/obspy#1802]
Sorry if the problem has been reported (I didn't find any on the issues list), but it seems inconsistent
the behaviour of this line:
https://github.com/obspy/obspy/blob/master/obspy/taup/tau_model.py#L238
where 
```
self._load_from_depth_cache()
```
is called without arguments (by providing the argument in the constructor `cache=False`)

Regardless of whether I am using the wrong `cache` value (which from the code it does not seem the case, but any help is appreciated in case), this seems to be a bug as the method expects an argument which is not given

Thank you very much in advance

**Comment by krischer** [comment]
Yea this does seem to be a bug. It likely never surfaces as is it fairly hard to initialize the `TauModel` class without a cache as all the high-level interfaces do set it. But we should eventually fix this nonetheless.

**Comment by rizac** [comment]
Yes indeed I tried some optimization digging into the code and I ended up to a very particular case. Anyway, from the code I really guess the problem is just replacing:
```
self._load_from_depth_cache()
```
with
```
self._load_from_depth_cache(depth)
```
(let alone of course writing missing tests)

**Comment by krischer** [comment]
I think so as well.

Please send a PR with any optimizations you end up discovering, best including some tests. `obspy.taup` is otherwise tested fairly well so any change that does not break the tests is likely valid.

**Comment by rizac** [comment]
Created PR #1805 which has also been merged with #1804 (see enhancement request #1803)
